packages/backend-function/src/function_env_translator.ts

```diff
@@ -43,6 +46,7 @@ export class FunctionEnvironmentTranslator {
       } else {
         this.lambda.addEnvironment(key, value);
       }
+      this.definedEnvVarNames.push(key);
```

Comment:
we need to append to this list in the `addSsmEnvironmentEntry` method as well

Coming back to this after reading the rest, I see this is intended to only be the env vars defined by the customer. Since `addSsmEnvironmentEntry` isn't necessarily just for resource access env vars maybe we should just group by "LambdaProvidedEnvVars" and "AmplifyBackendEnvVars". Or maybe we remove the grouping altogether and go back to a single type object

Reply:
Hmm `AmplifyBackendEnvVars` seems like a good name for the combined defined and resource access groups.

The naming for resource access env vars are somewhat derived by customer input right? Then we can group them together and customers should have an idea of where it came from and not confuse it with defined env vars.